front/app/containers/IdeasShow/GoBackButton.tsx

```diff
@@ -1,61 +1,92 @@
 import React, { memo } from 'react';
-import clHistory from 'utils/cl-router/history';
+
+// components
 import { Button, useBreakpoint, Box } from '@citizenlab/cl2-component-library';
+
+// hooks
 import useProject from 'hooks/useProject';
 import useLocale from 'hooks/useLocale';
-import useLocalize from 'hooks/useLocalize';
+
+// utils
+import clHistory from 'utils/cl-router/history';
 import { isNilOrError } from 'utils/helperUtils';
 import eventEmitter from 'utils/eventEmitter';
 import { ScreenReaderOnly } from 'utils/a11y';
 
+// i18n
+import messages from './messages';
+import { injectIntl, FormattedMessage } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+
 interface Props {
   projectId: string;
   className?: string;
   insideModal: boolean;
+  deselectIdeaOnMap?: () => void;
 }
 
-const GoBackButton = memo(({ projectId, className, insideModal }: Props) => {
-  const project = useProject({ projectId });
-  const locale = useLocale();
-  const isSmallTablet = useBreakpoint('smallTablet');
-  const localize = useLocalize();
-
-  const onGoBack = (event: React.MouseEvent) => {
-    event.preventDefault();
-
-    if (insideModal) {
-      eventEmitter.emit('closeIdeaModal');
-    } else if (!isNilOrError(project)) {
-      clHistory.push(`/projects/${project.attributes.slug}`);
-    } else {
-      clHistory.push('/');
+const GoBackButton = memo(
+  ({
+    projectId,
+    className,
+    insideModal,
+    deselectIdeaOnMap,
+  }: Props & InjectedIntlProps) => {
+    const project = useProject({ projectId });
+    const locale = useLocale();
+    const isSmallTablet = useBreakpoint('smallTablet');
+
+    const projectExists = !isNilOrError(project);
+    const deselectIdeaCallbackExists = !isNilOrError(deselectIdeaOnMap);
+
+    // a "back to ideas" message when returning to the list of ideas (e.g. on map)
+    // or a generic "go back message" when returning to the project main page
+    const buttonMessage =
+      projectExists && deselectIdeaCallbackExists
+        ? messages.backToIdeas
```

Comment:
You can have a different input term than idea (inside a project), so the copy needs to take this into account. You need to use `getInputTermMessage` for this.

Reply:
Got it. Is there a document or part of the codebase I can look at to see how that works in conjunction with our i18n system?

Reply:
If you search for getInputTermMessage, you'll find many examples. If these don't make it clear, DM me and I'll be glad to help/document.